plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/ArtifactHandler.kt

```diff
@@ -310,8 +314,25 @@ class ArtifactHandler(private val project: Project, private val clientAdaptor: G
                 )
             }
 
-            is DownloadFailureReason.OTHER, is DownloadFailureReason.INVALID_ARTIFACT -> {
-                // No notification, only chat update
+            is DownloadFailureReason.INVALID_ARTIFACT -> {
+                if (error.artifactType == TransformationDownloadArtifactType.CLIENT_INSTRUCTIONS) {
+                    notifyStickyWarn(
+                        message("codemodernizer.notification.warn.download_failed_client_instructions_expired"),
+                        CODE_TRANSFORM_TROUBLESHOOT_DOC_DOWNLOAD_EXPIRED,
+                    )
+                } else {
+                    notifyStickyWarn(
+                        message("codemodernizer.notification.warn.download_failed_invalid_artifact", artifactText),
+                        CODE_TRANSFORM_TROUBLESHOOT_DOC_DOWNLOAD_EXPIRED,
+                    )
+                }
+            }
+
+            is DownloadFailureReason.OTHER -> {
+                notifyStickyWarn(
+                    message("codemodernizer.notification.warn.download_failed_other.content", artifactText, error.errorMessage),
+                    CODE_TRANSFORM_TROUBLESHOOT_DOC_DOWNLOAD_ERROR_OVERVIEW,
+                )
```

Comment:
For these 2 download failure reasons above, I'm using the same logic that we use to show the equivalent chat message when the download fails